sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1518,6 +1520,10 @@ class GpuRunningWindowIterator(
       fixerIndexMap.values.foreach(_.close())
       saveLastParts(Array.empty)
       saveLastOrder(Array.empty)
+      maybeSplitIter.foreach(_.close())
+      maybeSplitIter = Iterator.empty
+      cachedBatch.foreach(_.close())
```

Comment:
Do we need to close ```cachedBatch``` here? When we process the next batch, the original batch will transfer to ```ret``` in ```readNextInputBatch``` and set to ```None```. And the in the end, where we have no coming batches, suppose it's still ```None``` for ```cachedBatch```.

Reply:
You are right, but i think better to have it here, if `cachedBatch` is None, this will be a noop.